lib/ms2ex/context/items/constant_stats.ex

```diff
@@ -0,0 +1,116 @@
+defmodule Ms2ex.Items.ConstantStats do
+  alias Ms2ex.{Item, Items, Storage}
+
+  def get(%Item{} = item, option_id, level_factor) do
+    constant_id = item.metadata.options.constant_id
+    options = Storage.Items.ConstantOptions.lookup(constant_id, item.rarity)
+
+    if options do
+      get_stats(item, options, option_id, level_factor)
+    else
+      get_default(item, %{}, option_id, level_factor)
+    end
+  end
+
+  defp get_stats(item, options, option_id, level_factor) do
+    %{stats: stats, special_stats: special_stats} = options
+
+    constant_stats = Enum.into(stats, %{}, &{&1.attribute, Items.Stat.build(&1)})
+
+    constant_stats =
+      Enum.into(special_stats, constant_stats, &{&1.attribute, Items.Stat.build(&1)})
+
+    # TODO Implement Hidden ndd (defense) and wapmax (Max Weapon Attack)
+
+    if level_factor > 50 do
+      get_default(item, constant_stats, option_id, level_factor)
+    else
+      constant_stats
+    end
+  end
+
+  defp get_default(item, constant_stats, option_id, level_factor) do
+    base_options = Ms2ex.Storage.Items.PickOptions.lookup(option_id, item.rarity)
+
+    if base_options do
+      process_options(item, constant_stats, base_options, level_factor)
+    else
+      constant_stats
+    end
+  end
+
+  defp process_options(item, constant_stats, base_options, level_factor) do
+    Enum.reduce(base_options.constants, constant_stats, fn constant_pick, acc ->
+      calc_script = get_calc_script(constant_pick.stat)
+
+      if calc_script do
+        process_stat(item, acc, constant_pick, calc_script, level_factor)
+      else
+        acc
+      end
+    end)
+  end
+
+  defp process_stat(item, constant_stats, pick, calc_script, level_factor) do
+    script =
+      case :luaport.spawn(:calc_item_values, "priv/scripts/Functions/calcItemValues") do
+        {:ok, script, _args} -> script
+        {:error, {:already_started, script}} -> script
+      end
```

Comment:
We should start the script outside `Enum.reduce/3` so that it is only started once.
We should also consider whether it's better to use a random/dynamic ID for the script instead of `:calc_item_values`.